autode/transition_states/base.py

```diff
@@ -191,11 +199,21 @@ def get_displaced_atoms_along_mode(calc, mode_number, disp_magnitude=1.0,
         s_atoms = deepcopy(atoms)
 
     mode_disp_coords = calc.get_normal_mode_displacements(mode_number)
+    coords = np.array([atom.coord for atom in s_atoms])
 
     assert len(s_atoms) == len(mode_disp_coords)
 
-    for i in range(len(s_atoms)):
-        s_atoms[i].translate(vec=disp_magnitude * mode_disp_coords[i, :])
+    disp_coords = coords.copy() + disp_factor * mode_disp_coords
+
+    # Ensure the maximum displacement distance of an atom by applying this
+    # displacement is below the threshold  by incrementing backwards in
+    # steps of 0.02 Å
+    while np.max(np.linalg.norm(coords - disp_coords, axis=1)) > max_disp:
```

Comment:
this seems very inefficient

Reply:
also "max_displacement" isn't what this is, its the total displacement. max displacement implies the displacement can be lower than this

Reply:
any suggestions for efficiency? Have modified slightly. Also I think it is the atomic rather than total displacement

```
>>> coords1 = np.array([[0., 0., 0.], [1., 0., 0.]])
>>> coords2 = np.array([[1., 0., 0.], [1., 0., 0.]])
>>> np.linalg.norm(coords1 - coords2, axis=1)
array([1., 0.])
```

Reply:
I'm not sure the efficiency is a huge problem, the major time taker is always gonna be the orca/gaussian/whatever.
Maybe iterate through each displacement, find the largest one, and scale them all to that?
Idk if that would be better though

Reply:
also my issue was with "max". Max implies it can be less, but there will be one atom that is always displaced that amount. Not sure if there's a clearer thing to name it, its probably ok

Reply:
because it's always incremented backwards I think it'll always be less than the max

Reply:
what do you think of the new version? less likely for an infinite loop..

Reply:
i guess its better, still not wild though
max sounds like it can be a lot less, but max is a cap. In this case max is a target, which I think is weird

Reply:
fair point. Not sure I have any better ideas tho

Reply:
maybe change the comment from "Ensure the maximum displacement is below the threshold" to somethign like "ensure the maximum displacement of any single atom is roughly equal to the threshold" but probably a bit nicer than that

Reply:
otherwise feel free to merge :)

Reply:
hmm, actually it could be a lot below - if `disp_factor` was small